actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -267,9 +267,13 @@ def handle_positional_args(controller_options, inner_options, args, result, path
                 path_params -= controller_options.keys
                 path_params -= result.keys
```

Comment:
I don’t know how often this code path is reached (inside an `if args.size < path_params_size` block), but the argument for saving an array allocation in 0cbec58ae48279ae5e9fdf6bbdbceb32183215dd might apply for lines 267 and 268 as well.

Just a thought: Doesn’t the refactoring from `foo -= bar.keys` to `bar.each { |key, _| foo.delete(key) }` smell like there’s a method missing on the `Array` class?